src/field_theory/ratpoly.lean

```diff
@@ -0,0 +1,391 @@
+  map_add' := of_fraction_ring_add,
+  map_mul' := of_fraction_ring_mul }
+
+/-- `ratpoly K` is the field of fractions of the polynomials over `K`. -/
+instance : is_fraction_ring (polynomial K) (ratpoly K) :=
+{ map_units := λ y, by rw ← of_fraction_ring_algebra_map;
+    exact (aux_equiv K).to_ring_hom.is_unit_map (is_localization.map_units _ y),
+  eq_iff_exists := λ x y, by rw [← of_fraction_ring_algebra_map, ← of_fraction_ring_algebra_map];
+    exact (aux_equiv K).injective.eq_iff.trans (is_localization.eq_iff_exists _ _),
+  surj := by { rintro ⟨z⟩, convert is_localization.surj (polynomial K)⁰ z, ext ⟨x, y⟩,
+    simp only [← of_fraction_ring_algebra_map, function.comp_app, ← of_fraction_ring_mul] } }
+
+variables {K}
+
+@[simp] lemma lift_on_div {P : Sort v} (p q : polynomial K)
+  (f : ∀ (p q : polynomial K), P) (f0 : ∀ p, f p 0 = f 0 1)
+  (H : ∀ {p q p' q'} (hq : q ≠ 0) (hq' : q' ≠ 0), p * q' = p' * q → f p q = f p' q') :
+  (algebra_map _ (ratpoly K) p / algebra_map _ _ q).lift_on f @H = f p q :=
+by rw [← mk_eq_div, lift_on_mk _ _ f f0 @H]
+
+/-- Induction principle for `ratpoly K`: if `f p q : P (p / q)` for all `p q : polynomial K`,
+then `P` holds on all elements of `ratpoly K`.
+
+See also `induction_on'`, which is a recursion principle defined in terms of `ratpoly.mk`.
+-/
+protected lemma induction_on {P : ratpoly K → Prop} (x : ratpoly K)
+  (f : ∀ (p q : polynomial K) (hq : q ≠ 0),
+    P (algebra_map _ (ratpoly K) p / algebra_map _ _ q)) :
+  P x :=
+x.induction_on' (λ p q hq, by simpa using f p q hq)
+
+lemma of_fraction_ring_mk' (x : polynomial K) (y : (polynomial K)⁰) :
+  of_fraction_ring (is_localization.mk' _ x y) = is_localization.mk' (ratpoly K) x y :=
+by rw [is_fraction_ring.mk'_eq_div, is_fraction_ring.mk'_eq_div, ← mk_eq_div', ← mk_eq_div]
+
+@[simp] lemma of_fraction_ring_eq :
+  (of_fraction_ring : fraction_ring (polynomial K) → ratpoly K) =
+    is_localization.alg_equiv (polynomial K)⁰ _ _ :=
+funext $ λ x, localization.induction_on x $ λ x,
+by simp only [is_localization.alg_equiv_apply, is_localization.ring_equiv_of_ring_equiv_apply,
+    ring_equiv.to_fun_eq_coe, localization.mk_eq_mk'_apply, is_localization.map_mk',
+    of_fraction_ring_mk', ring_equiv.coe_to_ring_hom, ring_equiv.refl_apply, set_like.eta]
+
+@[simp] lemma to_fraction_ring_eq :
+  (to_fraction_ring : ratpoly K → fraction_ring (polynomial K)) =
+    is_localization.alg_equiv (polynomial K)⁰ _ _ :=
+funext $ λ ⟨x⟩, localization.induction_on x $ λ x,
+by simp only [localization.mk_eq_mk'_apply, of_fraction_ring_mk', is_localization.alg_equiv_apply,
+  ring_equiv.to_fun_eq_coe, is_localization.ring_equiv_of_ring_equiv_apply, is_localization.map_mk',
+  ring_equiv.coe_to_ring_hom, ring_equiv.refl_apply, set_like.eta]
+
+@[simp] lemma aux_equiv_eq :
+  aux_equiv K = (is_localization.alg_equiv (polynomial K)⁰ _ _).to_ring_equiv :=
+by { ext x,
+     simp only [aux_equiv, ring_equiv.coe_mk, of_fraction_ring_eq, alg_equiv.coe_ring_equiv'] }
+
```

Comment:
Should we also have definitions like `ratpoly.X` and `ratpoly.C`? The alternative for now is to write `algebra_map _ _ polynomial.X`.

Reply:
I certainly think that it's good to have dedicated `X` and `C` + a bit of API.